lib/matplotlib/axes/_axes.py

```diff
@@ -6061,6 +6061,46 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
         bin_range = range
         del range
```

Comment:
There is a namespace conflict here. `range` is both the range of histogram and the python function. If I do just do a quick and dirty example:
```
del range
range(np.arange(100))
```
I get the below error message:
```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-21-0eb074616bf8> in <module>()
----> 1 del range
      2 range(np.arange(100))
NameError: name 'range' is not defined
```

This seems similar to the errors generated in the tests, which comes from using `range` to refer to the built-in python function on line 6171.